homeassistant/components/utility_meter/sensor.py

```diff
@@ -153,11 +157,27 @@ def __init__(
         self._tariff = tariff
         self._tariff_entity = tariff_entity
 
+    def start(self, unit):
+        """Initialize unit and state upon source initial update."""
+        self._unit_of_measurement = unit
+        self._state = 0
+        self.async_write_ha_state()
+
     @callback
     def async_reading(self, event):
         """Handle the sensor state changes."""
         old_state = event.data.get("old_state")
         new_state = event.data.get("new_state")
+
+        if (
+            self._unit_of_measurement is None or self._state is None
+        ) and new_state.state:
```

Comment:
This changes the behavior to always reset the meter if the state is `None` for unitless meters instead of just ignoring a `None`-state.
Let's add a new attribute, `self._started` instead, and set that to `True` both in the `start` method and if `self.async_get_last_state()` returns a valid state.

Reply:
Sorry, but that is the guard

If any of them is None (maybe I can change to **and**) that means the utility_meter sensor is going through its very first initialisation (afterwards restore state always guarantees this never happens)